osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -68,16 +70,38 @@ public SampleChannel GetSample(ISampleInfo sampleInfo)
 
         public IBindable<TValue> GetConfig<TLookup, TValue>(TLookup lookup)
         {
-            if (AllowConfigurationLookup && skin != null)
+            if (skin != null)
             {
-                var bindable = skin.GetConfig<TLookup, TValue>(lookup);
-                if (bindable != null)
-                    return bindable;
+                switch (lookup)
+                {
+                    // todo: the GlobalSkinColours switch is pulled from LegacySkin and should not exist.
+                    // will likely change based on how databased storage of skin configuration goes.
+                    case GlobalSkinColours global:
+                        switch (global)
+                        {
+                            case GlobalSkinColours.ComboColours:
+                                return getBindable<TLookup, TValue>(lookup, AllowColourLookup);
+                        }
+
+                        break;
+
+                    default:
+                        return getBindable<TLookup, TValue>(lookup, AllowConfigurationLookup);
+                }
             }
 
             return fallbackSource?.GetConfig<TLookup, TValue>(lookup);
         }
 
+        private IBindable<TValue> getBindable<TLookup, TValue>(TLookup lookup, bool bindableReturnCheck)
+        {
+            var bindable = skin.GetConfig<TLookup, TValue>(lookup);
+            if (bindable != null && bindableReturnCheck)
+                return bindable;
+            else
+                return fallbackSource?.GetConfig<TLookup, TValue>(lookup);
+        }
```

Comment:
naming's a little obtuse here, would probably do

```suggestion
        private IBindable<TValue> lookupWithFallback<TLookup, TValue>(TLookup lookup, bool canUseSkinLookup)
        {
            var bindable = skin.GetConfig<TLookup, TValue>(lookup);
            if (bindable != null && canUseSkinLookup)
                return bindable;
            else
                return fallbackSource?.GetConfig<TLookup, TValue>(lookup);
        }
```

Reply:
Yeah, I struggled to try to come up with good naming for this. I like your suggested naming much more than what I had.